core/templates/pages/contributor-dashboard-page/opportunities-list/opportunities-list.component.ts

```diff
@@ -115,25 +116,33 @@ export class OpportunitiesListComponent {
     this.directiveSubscriptions.unsubscribe();
   }
 
+
   ngOnInit(): void {
     this.loadingOpportunityData = true;
     this.activePageNumber = 1;
-    this.loadOpportunities().then(({opportunitiesDicts, more}) => {
-      // This ngZone run closure will not be required after \
-      // migration is complete.
-      this.zone.run(() => {
-        this.opportunities = opportunitiesDicts;
-        this.more = more;
-        this.visibleOpportunities = this.opportunities.slice(
-          0, this.OPPORTUNITIES_PAGE_SIZE);
-        this.userIsOnLastPage = this.calculateUserIsOnLastPage(
-          this.opportunities,
-          this.OPPORTUNITIES_PAGE_SIZE,
-          this.activePageNumber,
-          this.more);
-        this.loadingOpportunityData = false;
+
+    this.fetchAndLoadOpportunities();
+  }
+
+  fetchAndLoadOpportunities(): void {
+    if (this.loadOpportunities) {
+      this.loadOpportunities().then(({opportunitiesDicts, more}) => {
+        // This ngZone run closure will not be required after \
+        // migration is complete.
+        this.zone.run(() => {
+          this.opportunities = opportunitiesDicts;
+          this.more = more;
+          this.visibleOpportunities = this.opportunities.slice(
+            0, this.OPPORTUNITIES_PAGE_SIZE);
+          this.userIsOnLastPage = this.calculateUserIsOnLastPage(
+            this.opportunities,
+            this.OPPORTUNITIES_PAGE_SIZE,
+            this.activePageNumber,
+            this.more);
+          this.loadingOpportunityData = false;
+        });
       });
-    });
+    }
```

Comment:
Can we flip this check to use a guard clause to reduce nesting?

Reply:
Added!